src/context/WalletContext.tsx

```diff
@@ -38,6 +39,17 @@ export const WalletContextProvider = ({
     utxos,
   };
 
+  useEffect(() => {
+    if (
+      Cookies.get('wallet-connected') === 'true' &&
+      window.ergo_request_read_access
+    ) {
+      window
+        .ergo_request_read_access()
+        .then((flag) => setIsWalletConnected(flag));
+    }
+  }, [isWalletConnected]);
```

Comment:
In this case, a wallet cannot be disconnected because this code connects it again.
I think a better solution is to introduce `walletConnectionState: 'notConnected' | 'connected' | 'disconnected'` where `notConnected` indicates the initial state.
And then condition above can be rewritten with
```ts
  walletConnectionState === 'not_connected' &&
      Cookies.get('wallet-connected') === 'true' &&
      window.ergo_request_read_access
```